src/user_input_compiler.rs

```diff
@@ -825,14 +825,7 @@ mod test {
   #[test]
   fn webcam_42() {
     match Input::from( format!("/webcam 42") ) {
-      WebCam{ nth } => {
-        if let Some(nth) = nth { 
-          assert!( if nth == 42 { true } else { println!("wrong Value: {}", nth); false } ) 
-        } else {
-          println!("expected OptionalInteger");
-          assert!(false)
-        }
-      },
+      WebCam{ nth } => assert_eq!(Some(42), nth),
```

Comment:
I think you could reduce it even further to `assert_eq!(WebCam{ nth=Some(42)}, Input::from(...))`. See [here](https://play.rust-lang.org/?code=%0A%23%5Bderive(Debug%2C%20PartialEq%2C%20Eq)%5D%0Astruct%20Foo%20%7B%0A%20%20%20%20i%3A%20i32%2C%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20foo%20%3D%20Foo%7Bi%3A%2010%7D%3B%0A%20%20%20%20assert_eq!(foo%2C%20Foo%7Bi%3A%2010%7D)%3B%0A%7D%0A&version=stable).